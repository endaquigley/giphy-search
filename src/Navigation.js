import React, { Fragment, useContext } from "react";
import styled from "styled-components";

import { Store } from "./store";
import * as actions from "./actions";

const Heading = styled.h1`
  font-size: 50px;
  line-height: 1;
  margin-top: 0;
  margin-bottom: 15px;
`;

const Button = styled.button`
  overflow: hidden;
  margin: 5px;
  padding: 12px 12px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  transition: all 150ms linear;
  text-align: center;
  white-space: nowrap;
  text-decoration: none !important;
  border: 0 none;
  border-radius: 2px;
  font-size: 13px;
  font-weight: 500;
  line-height: 1.3;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  color: #202129;
  background-color: #f2f2f2;

  &:hover:not(:disabled) {
    opacity: 1;
    color: #202129;
    background-color: #e1e2e2;
  }

  &:disabled {
    opacity: 0.6;
  }
`;

const Container = React.memo(() => {
  const {
    state: { query },
    dispatch
  } = useContext(Store);

  const updateQuery = query => {
    return dispatch(actions.updateQuery(query));
  };

  return <Navigation query={query} updateQuery={updateQuery} />;
});

export const Navigation = React.memo(({ query, updateQuery }) => (
  <Fragment>
    <Heading>{query === "dog" ? "🐶" : "🐱"}</Heading>
    <Button onClick={() => updateQuery("dog")} disabled={query === "dog"}>
      Dog Search
    </Button>
    <Button onClick={() => updateQuery("cat")} disabled={query === "cat"}>
      Cat Search
    </Button>
  </Fragment>
));

export default Container;
