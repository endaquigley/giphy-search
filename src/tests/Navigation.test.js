import React from "react";
import { render } from "react-testing-library";

import { Navigation } from "../Navigation";

import "react-testing-library/cleanup-after-each";

describe("Navigation Component", () => {
  it("should render in the cat state", () => {
    const { container, getByText } = render(
      <Navigation query="cat" updateQuery={jest.fn()} />
    );

    expect(getByText("Dog Search").disabled).toBe(false);
    expect(getByText("Cat Search").disabled).toBe(true);

    expect(container).toMatchSnapshot();
  });

  it("should render in the dog state", () => {
    const { container, getByText } = render(
      <Navigation query="dog" updateQuery={jest.fn()} />
    );

    expect(getByText("Dog Search").disabled).toBe(true);
    expect(getByText("Cat Search").disabled).toBe(false);

    expect(container).toMatchSnapshot();
  });
});
