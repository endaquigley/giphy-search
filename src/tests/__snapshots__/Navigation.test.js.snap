// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component should render in the cat state 1`] = `
<div>
  <h1
    class="sc-bdVaJa hMOmBt"
  >
    🐱
  </h1>
  <button
    class="sc-bwzfXH bwQeco"
  >
    Dog Search
  </button>
  <button
    class="sc-bwzfXH bwQeco"
    disabled=""
  >
    Cat Search
  </button>
</div>
`;

exports[`Navigation Component should render in the dog state 1`] = `
<div>
  <h1
    class="sc-bdVaJa hMOmBt"
  >
    🐶
  </h1>
  <button
    class="sc-bwzfXH bwQeco"
    disabled=""
  >
    Dog Search
  </button>
  <button
    class="sc-bwzfXH bwQeco"
  >
    Cat Search
  </button>
</div>
`;
